app/core/Tracker2/DriftTracker.js

```diff
@@ -66,21 +63,25 @@ export default class DriftTracker extends BaseTracker {
     )
 
     this.watchForDisableAllTrackingChanges(this.store)
+  }
 
-    const isChina = (window.features || {}).china
-    if (isChina) {
-      this.enabled = false
-    }
-    else {
-      this.enabled = true
-    }
-    await this.initializationComplete
-    this.updateDriftConfiguration()
+  async initializeDrift () {
+    loadDrift()
 
-    const retries = await getPageUnloadRetriesForNamespace('drift')
-    for (const retry of retries) {
-      this[retry.identifier](...retry.args)
-    }
+    window.drift.on('ready', async (api) => {
+      this.driftApi = api
+
+      this.initDriftOnLoad()
+      if (!this.initializationComplete)
+        this.onInitializeSuccess()
```

Comment:
does this logic theck if we have `initializationComplete` promise?

Reply:
if we want to run `initializationComplete` (or get this promise result) we need await in async function i think

Reply:
I think my intention was not to wait for it, but just to see whether it exists, in progress or not. Does that make sense?